dataset-processor-python/data_loader.py

```diff
@@ -1,55 +1,272 @@
+import json
 import os
+import re
 import sys
 import csv
+import pandas as pd
+import numpy
 
 
-class DataLoader:
+class UserInputHandler:
+    """
+    Class to handle user inputs
+    """
+
     def __init__(self, path, dataset_name):
         self.path = path
         self.dataset_name = dataset_name
-        self.initial_data = None
+        self.inputs = self.get_user_inputs()
+
+    @staticmethod
+    def is_valid_version(version):
+        pattern = r"^\d{4}\.[0-9]+$"
+        return re.match(pattern, version) is not None
+
+    def get_user_inputs(self):
+        version = input("Enter the version: ").strip()
+        if not version or not self.is_valid_version(version):
+            print("Please enter a valid version in the format [yyyy.number].")
+            return self.get_user_inputs()
+        title = input("Enter the title: ").strip()
+        description = input("Enter the description: ").strip()
+        return {
+            "path": self.path,
+            "dataset_name": self.dataset_name,
+            "version": version,
+            "title": title,
+            "description": description,
+        }
+
+
+class DataLoader:
+    """
+    Loads the csv file and stores the initial data
+    """
+
+    def __init__(self, inputs):
+        self.inputs = inputs
+        self.data = self.load_csv()
 
-    def read_csv(self):
+    def load_csv(self):
         """
-        read the csv file and store the data
+        Read the csv file and store the data
         """
         try:
-            with open(self.path, "r") as f:
+            with open(self.inputs["path"], "r") as f:
                 reader = csv.DictReader(f)
-                data = list(
-                    reader
-                )  # data is a list of dictionaries, each dictionary is a row
```

Comment:
This comment was good and should be included in the method docstring! It's not clear to me what `load_csv` does just from the name or current description.